src/libraries/System.Security.Cryptography/tests/X509Certificates/PublicKeyTests.cs

```diff
@@ -746,6 +747,27 @@ public static void ExportSubjectPublicKeyInfo_ECDH()
             Assert.Equal(0, written);
         }
 
+        [ConditionalTheory(typeof(MLKem), nameof(MLKem.IsSupported))]
+        [MemberData(nameof(MLKemAlgorithms))]
+        public static void ExportSubjectPublicKeyInfo_MLKem(MLKemAlgorithm algorithm)
+        {
+            using MLKem kem = MLKem.GenerateKey(algorithm);
+            PublicKey key = new(kem);
+
+            Span<byte> algSpki = kem.ExportSubjectPublicKeyInfo();
+            AssertExtensions.SequenceEqual(algSpki, key.ExportSubjectPublicKeyInfo().AsSpan());
+
+            // Just right
+            algSpki.Clear();
+            Assert.True(key.TryExportSubjectPublicKeyInfo(algSpki, out int written), nameof(key.TryExportSubjectPublicKeyInfo));
+            Assert.Equal(algSpki.Length, written);
+            AssertExtensions.SequenceEqual(algSpki, key.ExportSubjectPublicKeyInfo().AsSpan());
+
+            // Too small
+            Assert.False(key.TryExportSubjectPublicKeyInfo(algSpki.Slice(1), out written), nameof(key.TryExportSubjectPublicKeyInfo));
+            Assert.Equal(0, written);
```

Comment:
Too big, mainly just to ensure that written doesn't report input.Length?

The array-returning one might also warrant either a NotSame across two calls, or that calling it, messing with the array/span, and calling it again yields a different answer (which happens to be the original answer, as already proven)... just showing that caller-contamination doesn't mutate the instance.